src/main/java/duke/helper/Parser.java

```diff
@@ -0,0 +1,82 @@
+package duke.helper;
+
+import duke.exception.EmptyTaskException;
+import duke.exception.InvalidCommandException;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.ToDo;
+
+public class Parser {
+    /**
+     * Validates the task input, checks for empty tasks. If task is empty, throws EmptyTaskException.
+     *
+     * @param input the user input.
+     * @throws EmptyTaskException
+     */
+    public void validateTaskInput(String input) throws EmptyTaskException {
+        String s = input.strip();
+        if (s.equals("todo") || s.equals("deadline") || s.equals("event")) {
+            throw new EmptyTaskException("Task " + s + " cannot be empty nya!\n");
+        }
+    }
+
+    /**
+     * Creates an appropriate task from the user input. Tasks created can be ToDo, Deadline or Event.
+     *
+     * @param input the user input.
+     * @return Task the resultant task formed.
+     */
+    public Task makeTaskFromInput(String input) {
+        validateTaskInput(input);
+        String rest = input.substring(input.indexOf(" ") + 1);
+        switch (input.split(" ")[0]) {
+        case "todo":
+            return new ToDo(rest);
+        case "deadline":
+            String[] words = rest.split("/by");
+            return new Deadline(words[0].strip(), words[1].strip());
+        default:
+            String description = rest.split("/from")[0].strip();
+            String start = rest.split("/from")[1].split("/to")[0].strip();
+            String end = rest.split("/to")[1].strip();
+            return new Event(description, start, end);
+        }
+    }
+
+    /**
+     * dispatches the commands based off the user input.
+     *
+     * @param command
+     * @param ui
+     * @param taskList
+     * @throws InvalidCommandException
+     */
+    public void dispatch(String command, Ui ui, TaskList taskList) throws InvalidCommandException {
+        switch (command.split(" ")[0]) {
```

Comment:
Would it be better if you took out `command.split(" ")[0]` and named it `command` instead? 
I see the constraint in the argument name but perhaps you could name it `userInput`.
I do see this issue in some of the rest of the conditional cases as well.

For [code quality](https://nus-cs2103-ay2223s2.github.io/website/se-book-adapted/chapters/codeQuality.html#guideline-maximize-readability), it would be great to _maximise readability_! 🧐